Introduce CMake workflow

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #15804

CMake workflow combines configure and build into 1 command. Instead of
doing:

```
cmake --preset llm \
      -DEXECUTORCH_BUILD_CUDA=ON \
      -DCMAKE_INSTALL_PREFIX=cmake-out \
      -DCMAKE_BUILD_TYPE=Release \
      -Bcmake-out -S.
cmake --build cmake-out -j$(nproc) --target install --config Release
```

We can simply do `cmake --workflow llm-release-cuda`. This largely
reduces the burden of running these cmake commands.

Next step I'm going to create workflow for the popular runners (llama,
whisper, voxtral etc) and further simplify the build command

larryliu0820 committed:
Introduce CMake workflow

CMake workflow combines configure and build into 1 command. Instead of
doing:

```
cmake --preset llm \
      -DEXECUTORCH_BUILD_CUDA=ON \
      -DCMAKE_INSTALL_PREFIX=cmake-out \
      -DCMAKE_BUILD_TYPE=Release \
      -Bcmake-out -S.
cmake --build cmake-out -j$(nproc) --target install --config Release
```

We can simply do `cmake --workflow llm-release-cuda`. This largely
reduces the burden of running these cmake commands.

Next step I'm going to create workflow for the popular runners (llama,
whisper, voxtral etc) and further simplify the build command

[ghstack-poisoned]

larryliu0820 committed:
Update on "Introduce CMake workflow"


CMake workflow combines configure and build into 1 command. Instead of
doing:

```
cmake --preset llm \
      -DEXECUTORCH_BUILD_CUDA=ON \
      -DCMAKE_INSTALL_PREFIX=cmake-out \
      -DCMAKE_BUILD_TYPE=Release \
      -Bcmake-out -S.
cmake --build cmake-out -j$(nproc) --target install --config Release
```

We can simply do `cmake --workflow llm-release-cuda`. This largely
reduces the burden of running these cmake commands.

Next step I'm going to create workflow for the popular runners (llama,
whisper, voxtral etc) and further simplify the build command

[ghstack-poisoned]

lucylq commented on tools/cmake/preset/README.md:
```diff
@@ -37,6 +37,47 @@ $ cmake -DEXECUTORCH_BUILD_MPS=OFF --preset llm
+- Uses `llm-release` configure preset (sets `CMAKE_BUILD_TYPE=Release`)
+- Uses `llm-release-install` build preset (builds the `install` target with parallel jobs)
+- Installs artifacts to `cmake-out/` directory
+
```
We can add some instruction on how to add a new worfklow. Is this correct? 

```
To add a new workflow:
1. Add a configure preset, e.g. new-workflow
2. Add a build preset that depends on (1), e.g. new-workflow-install
3. You should be able to run cmake --workflow new-workflow-install
```

larryliu0820 replied:
Yeah. Let me add your words into README.md

lucylq reviewed (approved):
This is a great change, thank you!

larryliu0820 committed:
Update on "Introduce CMake workflow"


CMake workflow combines configure and build into 1 command. Instead of
doing:

```
cmake --preset llm \
      -DEXECUTORCH_BUILD_CUDA=ON \
      -DCMAKE_INSTALL_PREFIX=cmake-out \
      -DCMAKE_BUILD_TYPE=Release \
      -Bcmake-out -S.
cmake --build cmake-out -j$(nproc) --target install --config Release
```

We can simply do `cmake --workflow llm-release-cuda`. This largely
reduces the burden of running these cmake commands.

Next step I'm going to create workflow for the popular runners (llama,
whisper, voxtral etc) and further simplify the build command

[ghstack-poisoned]
